server/routes/allocation-dashboard/handlers/endDateOption.ts

```diff
@@ -0,0 +1,49 @@
+import { Request, Response } from 'express'
+import { Expose } from 'class-transformer'
+import { IsNotEmpty } from 'class-validator'
+import { AllocationUpdateRequest } from '../../../@types/activitiesAPI/types'
+import ActivitiesService from '../../../services/activitiesService'
+
+export class EndDateOption {
+  @Expose()
+  @IsNotEmpty({ message: 'Choose whether you want to change or remove the end date.' })
+  endDateOption: string
+}
+
+export default class EndDateOptionRoutes {
+  constructor(private readonly activitiesService: ActivitiesService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { allocationId } = req.params
+    const allocation = await this.activitiesService.getAllocation(+allocationId, user)
+    const { scheduleId } = allocation
+    const { prisonerNumber } = allocation
+    res.render('pages/allocation-dashboard/end-date-option', {
+      scheduleId,
+      allocationId,
+      prisonerNumber,
+    })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    if (req.body.endDateOption === 'change') {
+      res.redirectOrReturn(`end-date`)
+    } else {
```

Comment:
I'd be tempted to remove this if else in favour of an early return. Eg, `if (req.body.endDateOption === 'change') return res.redirectOrReturn('end-date')`

Reply:
updated